client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -693,6 +693,9 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response) {
             log.debug("Received the following data from the controller {}", ranges.getSegmentRangesList());
             NavigableMap<Double, Segment> rangeMap = new TreeMap<>();
             for (SegmentRange r : ranges.getSegmentRangesList()) {
+                Preconditions.checkState(r.getMinKey() <= r.getMaxKey(),
```

Comment:
shouldnt this be strictly less than?

Reply:
Well in theory it is possible for a segment to be split so far that it extends to the maximum precision of a double and is left with nothing in one half... But that should not occur in practice.